sw/device/lib/aes.c

```diff
@@ -0,0 +1,116 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "aes.h"
+
+#include "aes_regs.h"
+#include "common.h"
+
+#define AES0_BASE_ADDR 0x40110000
+
+void aes_init(const aes_cfg_t aes_cfg) {
+  // Convert key_len from bytes to onehot
+  size_t key_len_onehot = 0b001;
+  if (aes_cfg.key_len_in_bytes == 24) {
+    key_len_onehot = 0b010;
+  } else if (aes_cfg.key_len_in_bytes == 32) {
+    key_len_onehot = 0b100;
+  }
+
+  REG32(AES_CTRL(0)) =
+      aes_cfg.mode << AES_CTRL_MODE |
+      (key_len_onehot & AES_CTRL_KEY_LEN_MASK) << AES_CTRL_KEY_LEN_OFFSET |
+      aes_cfg.manual_start_trigger << AES_CTRL_MANUAL_START_TRIGGER |
+      aes_cfg.force_data_overwrite << AES_CTRL_FORCE_DATA_OVERWRITE;
+  return;
+};
+
+void aes_key_put(const void *key, const size_t key_len_in_bytes) {
+  const size_t key_len_in_words = key_len_in_bytes >> 2;
+
+  for (int i = 0; i < key_len_in_words; i++) {
+    REG32(AES_KEY0(0) + i * sizeof(uint32_t)) = ((uint32_t *)key)[i];
+  }
+  for (int i = key_len_in_words; i < 8; i++) {
+    REG32(AES_KEY0(0) + i * sizeof(uint32_t)) = 0x0;
+  }
+
+  return;
```

Comment:
This return is not necessary; delete it; please do this throughout the file.

Reply:
Done.